### PR TITLE
feat: Add integration test runner

### PR DESCRIPTION
**Pull request by pksjce**
**What kind of change does this PR introduce?**
Adds scaffolding to chuck in test webpack configs.
Also add any fixes that break these config transformation.

**Did you add tests for your changes?**
Yep

**If relevant, did you update the documentation?**
Yep

**Summary**
Now all wild webpack v1 configs can be addes to the fixtures folder for integration testing.
If any snapshot does not work as expected, an issue can be raised for the relevant fix.

Required for #47 

**Does this PR introduce a breaking change?**
Nope

### CODE REVIEW

File: lib/transformations/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+const transform = require('./').transform;
+const no_of_files = 1;
+
+const runTest = (currentConfigPath) => {
+
```
Comment (okonet) [inline lib/transformations/integration.test.js]: Can you please remove this empty line?

File: lib/transformations/integration.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	let currentConfig = fs.readFileSync(currentConfigPath, 'utf8');
```
Comment (okonet) [inline lib/transformations/integration.test.js]: Should be `const`?

File: lib/transformations/integration.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+const transform = require('./').transform;
+const no_of_files = 1;
```
Comment (okonet) [inline lib/transformations/integration.test.js]: Please let's stick with `camelCased` convention for variable names.

File: lib/transformations/loaderOptionsPlugin/loaderOptionsPlugin.js [hunk 4]
```diff
@@ -19,7 +20,7 @@ module.exports = function(j, ast) {
 
 	return ast
 		.find(j.ArrayExpression)
-		.filter(path => path.parent.value.key.name === 'plugins')
+		.filter(path => safeTraverse(path, ['parent', 'value', 'key', 'name']) === 'plugins')
```
Comment (okonet) [inline lib/transformations/loaderOptionsPlugin/loaderOptionsPlugin.js]: 👍

File: lib/transformations/integration.test.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+const runTest = (currentConfigPath) => {
+
+	let currentConfig = fs.readFileSync(currentConfigPath, 'utf8');
+	const outputConfig = transform(currentConfig);
+	expect(outputConfig).toMatchSnapshot();
+};
+
+describe('Run integration tests', () => {
+	for(let i =1; i<= no_of_files; i++){
```
Comment (okonet) [inline lib/transformations/integration.test.js]: Please fix formatting here. I'm wondering why lint-staged didn't fix it though.

File: lib/transformations/integration.test.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+		test(`run test - webpack${i}.config.js`, () => {
```
Comment (okonet) [inline lib/transformations/integration.test.js]: Why use `it` everywhere. Let's switch for consistency.

File: lib/transformations/integration.test.js (hunk 5 above)
Comment (okonet) [inline lib/transformations/integration.test.js]: I'm not sure how is this suppose to work... `no_of_files` is set to 1 so the loop will always be from 0 to 1.

Comment (pksjce) [inline lib/transformations/integration.test.js]: Not sure how to pick up the test files. I took a very naive approach. Should I use regex for this then?

**Comment by pksjce** [comment]
Have separated the fix to it's own PR. Will close this one and create a new PR in some time.

**Pull request closed without merging**